### PR TITLE
Contracts added to CLI, tests updated. AEA-391

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

Contracts added to CLI functionality (add, search, push, etc.)
Tests updated.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by cyenyxe** [comment]
Is this still WIP even though a review has been requested? The code coverage decrease in `search.py` is pretty high (more than 10%), and the test file associated with it doesn't seem to have been updated.

File: aea/cli/search.py [hunk 1]
```diff
@@ -139,6 +140,38 @@ def connections(ctx: Context, query):
     print(format_items(sorted(result, key=lambda k: k["name"])))
 
 
+@search.command()
+@click.option("--query", default="", help="Query string to search Contracts by name.")
+@pass_ctx
+def contracts(ctx: Context, query):
+    """Search for Contracts."""
+    if ctx.config.get("is_registry"):
```
Comment (DavidMinarsch) [inline aea/cli/search.py]: Please refactor this and the other search commands do avoid code duplication and allow for code reuse.

Comment (DavidMinarsch) [reply aea/cli/search.py]: Rows 148-172 are generic apart from the `contracts` string afaikt

Comment (DavidMinarsch) [reply aea/cli/search.py]: Could have something like `_search_item(ctx, query, pluralized_item_type)`

File: aea/cli/common.py [hunk 2]
```diff
@@ -248,29 +252,6 @@ def format_items(items):
...
-def format_skills(items):
-    """Format list of skills to a string for CLI output."""
-    list_str = ""
-    for item in items:
-        list_str += (
-            "{line}\n"
-            "Public ID: {public_id}\n"
-            "Name: {name}\n"
-            "Description: {description}\n"
-            "Protocols: {protocols}\n"
```
Comment (DavidMinarsch) [inline aea/cli/common.py]: Note, @panasevychol you removed the dependency info from the search command output for skills now. That's ok but we should open an issue that dependencies should be shown for skills (protocols, contracts) and agents (skills, contracts, protocols, connections).

File: aea/cli/search.py [hunk 3]
```diff
@@ -103,121 +103,136 @@ def _get_details_from_dir(
...
+            configs[item_type_plural]["loader"],
+            registry,
+            "*/{}".format(item_type_plural),
+            configs[item_type_plural]["config_file"],
+            result,
+        )
+
+    _get_details_from_dir(
+        configs[item_type_plural]["loader"],
+        lookup_dir,
```
Comment (DavidMinarsch) [inline aea/cli/search.py]: I would change here `lookup_dir` to registry.

File: aea/cli/search.py [hunk 4]
```diff
@@ -103,121 +103,136 @@ def _get_details_from_dir(
...
+        lookup_dir = registry
+    else:
+        lookup_dir = AEA_DIR
+        _get_details_from_dir(
+            configs[item_type_plural]["loader"],
+            registry,
+            "*/{}".format(item_type_plural),
+            configs[item_type_plural]["config_file"],
+            result,
+        )
```
Comment (DavidMinarsch) [inline aea/cli/search.py]: I'd change here to:
```suggestion
    if item_type_plural != "agents":
        _get_details_from_dir(
            configs[item_type_plural]["loader"],
            AEA_DIR,
            "*/{}".format(item_type_plural),
            configs[item_type_plural]["config_file"],
            result,
        )
```

Comment (DavidMinarsch) [reply aea/cli/search.py]: plus change in following comment make this bit easier to read

File: aea/cli/search.py (hunk 3 above)
Comment (panasevychol) [reply aea/cli/search.py]: That's a bit different cause we look for items in registry and AEA_DIR, and for agents only in registry. Looks at those 2 calls.

File: aea/cli/search.py (hunk 4 above)
Comment (panasevychol) [reply aea/cli/search.py]: Not a big deal, can change. I just prefer in cases where we have one condition and everything else goes under `else` to check that very condition in `if` statement. But I won't insist.

File: aea/cli/common.py (hunk 2 above)
Comment (panasevychol) [reply aea/cli/common.py]: @DavidMinarsch we discussed that before and you told that we won't need protocols to be shown in skills search results. That just wasn't done that time.
IMO we should have an opened issue about output format. Now it looks a bit poor, maybe we should update it, add more info, better formatting. Will create one.

**Pull request merged**